container.go

```diff
@@ -624,3 +645,64 @@ func (container *container) unregisterCallback() error {
 
 	return nil
 }
+
+// Modifies the System by sending a request to HCS
+func (container *container) Modify(config *ResourceModificationRequestResponse) error {
+	operation := "Modify"
+	title := "HCSShim::Container::" + operation
+
+	if container.handle == 0 {
+		return makeContainerError(container, operation, "", ErrAlreadyClosed)
+	}
+
+	container.handleLock.RLock()
+	defer container.handleLock.RUnlock()
+
+	requestJSON, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+
+	var (
+		resultp *uint16
+	)
+
+	requestString := string(requestJSON)
+	logrus.Debugf(title+" id=%s request=%s", container.id, requestString)
+
+	err = hcsModifyComputeSystem(container.handle, requestString, &resultp)
+	err = processHcsResult(err, resultp)
+	if err != nil {
+		return makeContainerError(container, operation, "", err)
+	}
+	logrus.Debugf(title+" succeeded id=%s", container.id)
+	return nil
+}
+
+// ModifyContainer corresponding to the container id, by sending a request
+func ModifyContainer(id string, request *ResourceModificationRequestResponse) error {
```

Comment:
Yes, tested in RS1. Works fine.
# RS1
## Scenario: RS1 Connect/Disconnect Network on Non-Running Container - Pass
_C:\docker>.\docker create --name=test --rm microsoft/nanoserver ping 127.0.0.1 -t
6287bd48b1660b1ba65d4cccfb98cb2270f197efcb1f8c2125a976689e41871e
C:\docker>.\docker network connect trans test
C:\docker>.\docker start test
test_

## Scenario: RS1 Connect Network on Running Container return Unsupported error - PASS
_C:\docker>.\docker network connect trans test
Error response from daemon: unsupported platform request_

## Scenario: RS1 Disconnect Network on Running Container return Unsupported error - PASS-FAIL
_Platform throw unsupported request. Docker treats this as Warning & calls RemoveEndpoint directly, which detaches the end point. So it hot detach works, but unexpected. We should be okay_

# Current
## Scenario: [Current] Connect/Disconnect Network on Non-Running Container - Pass
## Scenario: [Current] Connect/Disconnect Network on Running Container  - PASS